### DISCUSSION TITLE
Adding extraAttributes to menu items?

### DISCUSSION THREAD

**Discussion by tiran133** [filamentphp/filament#12301] [Help]
### Package

Panel builder

### Package Version

latest

### How can we help you?

Hi,
im trying to add extra attributes to a menu item within ->userMenuItems([]) so I can call wire:click.

But I also would love to see this on `use Filament\Navigation\NavigationItem;`

Almost all Elements allow for extraAttributes() except this one.

For now Im doing some html tag injection, which feels very hacky.

```php
MenuItem::make()
    ->label(fn () => __('app.dashboard.menu_item_banner_change'))
    ->url('#" onclick="Livewire.dispatch(\'open-modal\',{id:\'chooseImageModal\'})')
    ->visible(fn () => auth()->user()->can('view Dashboard'))
    ->icon('heroicon-o-pencil'),
```

 Would love to see something like this

```php
MenuItem::make()
      ->label(fn () => __('app.dashboard.menu_item_banner_change'))
      ->url('#')
      ->extraAttributes([
          'wire:click.prevent' => "dispatch('open-modal',{id:'chooseImageModal'})",
      ])
      ->visible(fn () => auth()->user()->can('view Dashboard'))
      ->icon('heroicon-o-pencil'),
```

Kind regards

**Comment by bencarrr** [comment]
Id love to see this for `NavigationItem` too!
